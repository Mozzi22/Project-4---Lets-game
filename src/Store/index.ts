import { combineReducers } from 'redux';
import { reducer as registrationReducer } from './registration/reducer';
import { reducer as loginReducer } from './login/reducer';
import { reducer as modalReducer } from './modals/reducer';
import { reducer as gameReducer } from './games/reducer';

const rootReducer = combineReducers({
  registration: registrationReducer,
  login: loginReducer,
  game: gameReducer,
  modals: modalReducer,
});

export default rootReducer;
