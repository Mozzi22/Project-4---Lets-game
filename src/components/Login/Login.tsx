import React from 'react';
import { NavLink, Redirect } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { logInputs } from '../../constants/componentsÐ¡onsts';
import { APP_ROUTES } from '../../constants/reactRoutes';
import { StLoginDiv } from './styled';
import Form from '../UI/Form';
import Input from '../UI/Input';
import Button from '../UI/Button';
import { TInitialLoginData, TLoginData, TLoginValue } from '../../Store/login/types';

interface ILogin {
    sendLoginRequest: (e: React.MouseEvent) => void ;
    setLoginValue: (data: TLoginData) => void ;
    fields: TInitialLoginData;
    setValue: (data: TLoginValue) => void;
}

const Login = ({ sendLoginRequest, setLoginValue, fields, setValue }: ILogin) => {
  const { t } = useTranslation();
  if (fields.success) {
    setValue({ name: 'success', value: false });
    return <Redirect to={APP_ROUTES.registration} />;
  }
  return (
    <StLoginDiv>
      <Form>
        <img src="./public/assets/images/sign-logo.svg" alt='sign in' />
        <p>{t('authorization')}</p>
        {logInputs.map(input => (
          <Input
            width="400px"
            id={input.id}
            height="auto"
            key={input.id}
            name={input.id}
            padding="17px 15px"
            inputHeight="60px"
            border="0px"
            borderRadius="7px"
            color="#484b54"
            fontSizeInp="18px"
            label={t(input.label)}
            value={fields[input.id]}
            onChange={setLoginValue}
            placeholder={t(input.placeholder)}
          />
        ))}
        <Button
          height="50px"
          content="login_btn"
          id="login"
          onClick={sendLoginRequest}
        />
        <span>
          {t('login_span_text')}
          <NavLink to={APP_ROUTES.registration}>{t('login_link_text')}</NavLink>
        </span>
      </Form>
    </StLoginDiv>
  );
};

export default Login;
