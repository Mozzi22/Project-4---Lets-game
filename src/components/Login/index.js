import { connect } from 'react-redux';
import Login from './Login.jsx';
import { setLoginValue, sendLoginRequest, setValue } from '../../Store/login/actions';
import { loginStore } from '../../Store/login/selectors';

const mapStateToProps = state => ({ fields: loginStore(state) });

const mapDispatchToProps = dispatch => ({
    setLoginValue: payload => dispatch(setLoginValue(payload)),
    sendLoginRequest: () => dispatch(sendLoginRequest()),
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
