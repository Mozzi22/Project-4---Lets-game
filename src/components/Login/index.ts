import { connect } from 'react-redux';
import { setLoginValue, sendLoginRequest, setValue } from 'src/Store/login/actions';
import { loginStore } from 'src/Store/login/selectors';
import Login from './Login';

const mapStateToProps = state => ({ fields: loginStore(state) });

const mapDispatchToProps = dispatch => ({
    setLoginValue: payload => dispatch(setLoginValue(payload)),
    sendLoginRequest: (payload) => dispatch(sendLoginRequest(payload)),
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
