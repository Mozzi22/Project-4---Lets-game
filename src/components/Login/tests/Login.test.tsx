import React from 'react';
import { shallowSmart, mountSmart } from 'src/helpers/tests/testHelper';
import Login from '../Login';
import 'src/i18n';

describe('Login', () => {
    let props;
    const sendLoginRequest = jest.fn();
    const setLoginValue = jest.fn();
    const fields = {
        login: '',
        password: '',
        success: null,
    };
    beforeEach(() => {
        props = {
            setLoginValue,
            sendLoginRequest,
            fields,
        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<Login {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Input')).toHaveLength(2);
    });
    it('should render button', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = mountSmart(<Login {...props} />);
        component.find('Button').simulate('click');
        expect(sendLoginRequest).toHaveBeenCalled();
    });
    it('should click change input', () => {
        const component = mountSmart(<Login {...props} />);
        component.find('input').first().simulate('change', { target: { name: 'login', value: 'loginValue' } });
        expect(setLoginValue).toHaveBeenCalledWith({ name: 'login', value: 'loginValue' });
    });
    it('should redirect', () => {
        props.fields = {
            success: true,
        };
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Redirect')).toHaveLength(1);
    });
});
