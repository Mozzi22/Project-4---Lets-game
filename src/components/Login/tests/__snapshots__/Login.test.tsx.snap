// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `"<div class=\\"styled__StLoginDiv-sc-13c43a5-0 cIxYSG\\"><div class=\\"styled__StFormDiv-sc-2jn9rk-0 bySPyt\\"><img src=\\"./public/assets/images/sign-logo.svg\\" alt=\\"sign in\\"/><p>Sign In</p><div width=\\"400px\\" height=\\"auto\\" class=\\"styled__StInputContainer-sc-153pik-3 dstpmx\\"><label for=\\"email\\" class=\\"styled__StLabel-sc-153pik-0 gnhVmo\\">Email</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" color=\\"#484b54\\" placeholder=\\"Login\\" class=\\"styled__StInput-sc-153pik-1 kHsRis\\"/></div><div width=\\"400px\\" height=\\"auto\\" class=\\"styled__StInputContainer-sc-153pik-3 dstpmx\\"><label for=\\"password\\" class=\\"styled__StLabel-sc-153pik-0 gnhVmo\\">Password</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" color=\\"#484b54\\" placeholder=\\"Password\\" class=\\"styled__StInput-sc-153pik-1 kHsRis\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-15f3n0p-0 dhiANH\\">Sign in</button><span>Don\`t have an account?<a href=\\"/registration\\">Sign Up</a></span></div></div>"`;
